bitextor/bitextor_split.py

```diff
@@ -162,7 +162,7 @@ def split_segments(text, splitter_unary_func, prune_type="words", prune_threshol
 
             if process_paragraphs:
                 try:
-                    paragraph_id = int(column[1]) + 1 # Start at 1
+                    paragraph_id = column[1] # Starts at 1 and includes total number of paragraphs
```

Comment:
Try-except should be removed or adapted using `re`:

```python
import re

pattern = re.compile("^([0-9]+):([0-9]+)$")
m = re.match(pattern, paragraph_id)

if m:
  # We don't need a try-except for int() since we're sure that they are integers because of the regex
  paragraph_id = int(m.group(1))
  # More control than before, so we could:
  total_paragraphs = int(m.group(2))

  if paragraph_id > total_paragraphs:
    logging.warning("Paragraph id > total paragraphs (bug?): %d > %d", paragraph_id, total_paragraphs)
    # Or even raise Exception()
else:
  raise Exception(f"Couldn't process document #{doc_idx}, sentence #{sent_idx}")
```

Reply:
I just updated the code with your suggestions